### PR TITLE
DDT-736: NotNull tag for mandatory fields in AbstractBookingTO

### PR DESCRIPTION
**Pull request by mp-assco**

### CODE REVIEW

**Review by nt-as — COMMENTED** [review]
We use the wrong name for the field (starting with an upper case).  Maybe rephrase them as "The attribute X is required" to avoid the need for a starting capital letter in the field name.

(I noticed this mistake existed prior to this commit, but lets clean it up now)

**Comment by mp-assco** [comment]
> We use the wrong name for the field (starting with an upper case). Maybe rephrase them as "The attribute X is required" to avoid the need for a starting capital letter in the field name.
> 
> (I noticed this mistake existed prior to this commit, but lets clean it up now)

Sounds good to me.

File: src/main/java/org/dcsa/bkg/model/transferobjects/BookingTO.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
   private LocationTO placeOfIssue;
 
   @Valid
-  @NotEmpty(message = "Commodities are required.")
+  @NotEmpty(message = "The property commodities is required.")
```
Comment (douradofrazer) [inline src/main/java/org/dcsa/bkg/model/transferobjects/BookingTO.java:20]: Shouldn't it be `attribute` here instead of `property`? to be consistent.

File: src/test/java/org/dcsa/bkg/model/transferobjects/BookingTOTest.java [hunk 2]
```diff
@@ -251,7 +306,7 @@ void testToVerifyCommoditiesIsRequired() {
     validBookingTO.setCommodities(null);
     Set<ConstraintViolation<BookingTO>> violations = validator.validate(validBookingTO);
     Assertions.assertTrue(
-        violations.stream().anyMatch(v -> "Commodities are required.".equals(v.getMessage())));
+        violations.stream().anyMatch(v -> "The property commodities is required.".equals(v.getMessage())));
```
Comment (douradofrazer) [inline src/test/java/org/dcsa/bkg/model/transferobjects/BookingTOTest.java:309]: Ditto

File: src/test/java/org/dcsa/bkg/model/transferobjects/BookingTOTest.java [hunk 3]
```diff
@@ -260,7 +315,7 @@ void testToVerifyCommoditiesCannotBeAnEmptyList() {
     validBookingTO.setCommodities(Collections.emptyList());
     Set<ConstraintViolation<BookingTO>> violations = validator.validate(validBookingTO);
     Assertions.assertTrue(
-        violations.stream().anyMatch(v -> "Commodities are required.".equals(v.getMessage())));
+        violations.stream().anyMatch(v -> "The property commodities is required.".equals(v.getMessage())));
```
Comment (douradofrazer) [inline src/test/java/org/dcsa/bkg/model/transferobjects/BookingTOTest.java:318]: Ditto

**Review by douradofrazer — CHANGES_REQUESTED** [review]

File: src/main/java/org/dcsa/bkg/model/transferobjects/BookingTO.java (hunk 1 above)
Comment (mp-assco) [reply src/main/java/org/dcsa/bkg/model/transferobjects/BookingTO.java]: you're probably right. Changed it to `attribute`.

**Review by douradofrazer — APPROVED** [review]
